### ISSUE TITLE
Add lane highlight to Speed Limits tool

### ISSUE THREAD

**Issue by aubergine10** [CitiesSkylinesMods/TMPE#682]
When user is hovering a speed limit icon, particularly in lane-wise mode which allows per-lane speed setting, it is sometimes unclear which lane it corresponds to.

![image](https://user-images.githubusercontent.com/1386719/74113500-6d5cc880-4b9c-11ea-9777-7e75e84e09c9.png)  ![image](https://user-images.githubusercontent.com/1386719/74113524-9e3cfd80-4b9c-11ea-9932-6bac4b51a625.png)

Could we pimp the relevant lane with a highlight to make it completely clear which lane is being altered?

The lane highlight would only be visible while the speed limit icon is hovered.

Suggestion: White highlight (like outgoing lanes on Lane Connector tool) but without the border -a subtle highlight might be sufficient.

**Comment by kvakvs** [comment]
This might take more thinking.
I.e. those overlay signs should behave more like UI elements with the following properties:
* Change rendering slightly when they become UI elements (optional)
* Change color/brightness slightly mouse is hovered above them. Render an outline maybe.
* Actively resist overlapping other signs, sliding away along the road segment vector if too many signs were squeezed together.

Now that users will perceive them as UI elements, also add the suggestion from this ticket with lane highlights.

**Comment by aubergine10** [comment]
I find overlapping speed signs to be hugely useful timesaver - I can click overlap and update both directions at same time. I use that little trick constantly.

I think all our overlays are going to need a few iterations to improve aesthetics and usability. But for now I think just getting lane highlight solves the biggest issue of it not being overtly clear which lane/direction is being affected (especially on roads with large numbers of lanes).

**Comment by aubergine10** [comment]
Refs: #33

**Comment by kianzarrin** [comment]
Oops I added lane highlight to parking tool by mistake!:
![Screenshot (489)](https://user-images.githubusercontent.com/26344691/74573985-ed16d880-4f8a-11ea-8369-c22c2d945b03.png)
![Screenshot (490)](https://user-images.githubusercontent.com/26344691/74573993-f0aa5f80-4f8a-11ea-8e98-694aef40488b.png)

Holding shift will highlight all the lanes. I don't know how to take a screenshot of that though. Its a nice visual feedback though.

**Comment by aubergine10** [comment]
Noce lol. I forgot that parking tool had Shift modifier to apply to route so that too will benefit from lane highlight! Keep it :)

**Comment by kianzarrin** [comment]
> 
> 
> Noce lol. I forgot that parking tool had Shift modifier to apply to route so that too will benefit from lane highlight! Keep it :)

I need to add tooltip hotkey
its blocked by #701

**Comment by aubergine10** [comment]
I'll stash my current work on #699 and test your PRs tonight.

**Comment by kianzarrin** [comment]
OK this is done:
![Screenshot (493)](https://user-images.githubusercontent.com/26344691/74593128-c57a4b80-5030-11ea-8bcc-7bf3f568afaa.png)

I will take more screenshots later. shift also works, per lane also works.

PS: I might add reset functionality. I can replace the 140KMH button to reset button with del hotkey.
There is not much space on that panel. I might also multiply the size of that panel by resolution or something.

**Comment by kvakvs** [comment]
Shrink icons a little and find some space, or place the button below, but please don't reduce the palette. That UI is due for rework, and the tool window will be redone.

**Comment by aubergine10** [comment]
The speed palette has always been huge and many users complain it eats too much screen space. Related issues: #299, #573, #406.

**Comment by kianzarrin** [comment]
This http://www.41post.com/3816/programming/unity-scaling-the-gui-based-on-the-screen-resolution may be a solution to resolution of the GUI window.

**Comment by kvakvs** [comment]
GUI resolution should be done somewhat globally in a tool class, and then applied to all GUI windows. It is out of scope of this issue.

**Comment by kianzarrin** [comment]
![Screenshot (495)](https://user-images.githubusercontent.com/26344691/74595009-404d6180-5045-11ea-9374-99fca3a3b608.png)
![Screenshot (496)](https://user-images.githubusercontent.com/26344691/74595010-44797f00-5045-11ea-8290-05b660f13124.png)

The scaling has not produced nice textures.
text fonts also need to be scaled.

I performed scaling using `TrafficmangerTool.AdaptWidth/Height()` 

I haven't played with `GUI.matrix = Matrix4x4.TRS(new Vector3(GUIsF.x,GUIsF.y,0),Quaternion.identity,GUIsF);`
